### ISSUE TITLE
Best practise query output?

### ISSUE THREAD

**Issue by larspett** [rekonstrukt/swedishbutterflies#10]
We were discussing how to filter sql query outputs in the best way, either by letting the query deliver a more extensive (filterable) output or to provide parameters which are then integrated in the query. Both alternatives are feasible but I am uncertain if there are best practise standards? What would you advise?

**Comment by mskyttner** [comment]
I think (wider more extensive) filterable output data would be a good start in this case. In a case with big(ger) data the  parameterized approach might be better as it could save response time considerably and avoid getting huge chunks of data. I think that we will like not need to worry about that in our situation. If this assumption holds, it would then be convenient to get more data up-front and then filter in a later stage (if needed) rathen than having to filter already when requesting data from the source (db)...  And if we need to worry about bigger data in the future we should then perhaps consider other options too, like using datastores like Apache Spark and friends.

If you agree we can try to use this approach ie "extensive filterable output" consistently in the package for all the methods that get data, where it makes sense? Not sure if it means a lot of refactoring of db queries or not?

**Comment by larspett** [comment]
I can pretty easily adjust this queries to get a more extensive output. One thing that I thought of is if we should allow the use of alternative data by setting an input data path (while keeping the current data as default)? That is, the graphing function could be used for non-dB data too

**Comment by mskyttner** [comment]
Yes, sounds good. As you say, it would be nice to have the plot functions take (any) data as an optional input parameters - making it possible and easy to use data from the db but not a strict requirement if you want to feed some other data to the plot functions.

**Comment by larspett** [comment]
Should I have a look at adding eg species, municipality, region,  province, county, is_visible, date, and possible other table fields to the sql query output to enable in function filtering before plots? I am thinking setting parameters so that we get the output we have now but allowing for filtering.

**Comment by rekonstrukt** [comment]
Would be real nice!

**Comment by larspett** [comment]
Carsten and I did this jointly now. We fed the histogram plot an external file with more extensive column content. It allows filtering now and I will check & update the in code sql to add more extensive content
